checkov/sast/checks/base_registry.py

```diff
@@ -4,21 +4,25 @@
 import os
 import yaml
 from typing import List, Any, Optional, Set, Dict
+
 from checkov.common.bridgecrew.check_type import CheckType
 from checkov.sast.checks.base_check import BaseSastCheck
 from checkov.common.checks.base_check_registry import BaseCheckRegistry
 from checkov.runner_filter import RunnerFilter
+from checkov.sast.checks.checks_parser import SastCheckParser
 from checkov.sast.consts import SastLanguages
 from checkov.common.checks_infra.registry import CHECKS_POSSIBLE_ENDING
 
 
 class Registry(BaseCheckRegistry):
     def __init__(self, checks_dir: str) -> None:
         super().__init__(report_type=CheckType.SAST)
-        self.rules: List[str] = []
+        self.rules: List[Dict[str, Any]] = []
         self.checks_dir = checks_dir
         self.logger = logging.getLogger(__name__)
+        self.parser = SastCheckParser()
         self.runner_filter: Optional[RunnerFilter] = None
+        self.temp_semgrep_rules_path = os.path.join(self.checks_dir, 'temp_semgrep_rules.yaml')
```

Comment:
I think you should separate the string to a parameter for the init with default value, so it could later be overrided if needed

Reply:
I'm actually going abroad for a week so in order to not block this PR, I changed from "requesting changes" to just "commented", so please still answer these comments :) No need to wait for me to return as long as you handle all of the comments

Reply:
Done